tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/CircuitBreakerTest.java

```diff
@@ -624,4 +627,98 @@ public void testClassLevelCircuitOverrideNoDelay() {
 
         Assert.assertEquals(serviceAExecutions, 7, "The number of executions should be 7");
     }
+    
+    /**
+     * A test to exercise Circuit Breaker rolling window
+     * 
+     * With requestVolumeThreshold = 4, failureRatio=0.5, expected behaviour is,
+     * 
+     * Execution Behaviour 
+     * ========= ========= 
+     * 1         Success 
+     * 2         RunTimeException 
+     * 3         RunTimeException  
+     * 4         Success 
+     * 5         CircuitBreakerOpenException 
+     */
+    @Test
+    public void testRollingWindowCircuitOpen() {
+        boolean cbExceptionThrown = false;
+        for (int i = 1; i < 6; i++) {
+
+            try {
+                clientForRW.serviceARollingWindowOpenAfter4();
+            }
+            catch (CircuitBreakerOpenException cboe) {
+                // Expected on iteration 5
+                cbExceptionThrown = true;
+                if (i != 5) {
+                    Assert.fail("serviceA should not throw a CircuitBreakerOpenException in testRollingWindowCircuitOpen on iteration " + i);
+                }
+            }
+            catch (RuntimeException ex) {
+                // Expected
+                if (!contains(new int[]{2, 3}, i)) {
+                    Assert.fail("serviceA should not throw a RuntimeException on iteration " + i);
+                }
+            }
+            catch (Exception ex) {
+                // Not Expected
+                Assert.fail("serviceA should throw a RuntimeException or CircuitBreakerOpenException in testRollingWindowCircuitOpen "
+                     + "on iteration " + i);
+            }
+        }
+        int serviceAExecutions = clientForRW.getCounterForInvokingServiceA();
+        Assert.assertTrue(cbExceptionThrown, "The CircuitBreaker exception in testRollingWindowCircuitOpen was not thrown correctly.");
+
+        Assert.assertEquals(serviceAExecutions, 5, "The number of executions should be 5");
```

Comment:
I think we expect 4 executions here, the fifth throws a CircuitBreakerOpenException and the method body is not executed.

Reply:
yep. thanks